### ISSUE TITLE
Enable-AzureRmContextAutosave fails with different PowerShell editions (Full and Core)

### ISSUE THREAD

**Issue by iricigor** [Azure/azure-powershell#6340]
### Description

Function **Enable-AzureRmContextAutosave** on Windows machines (regardless of Full or Core PowerShell) uses following file for saving token: `C:\Users\<Username>\AppData\Roaming\Windows Azure Powershell\TokenCache.dat`. 

This is creating problems as Full and Core versions are using **different mechanisms for login**. 

### Script/Steps for Reproduction

```powershell
# In Full PowerShell, run these commands
Enable-AzureRmContextAutosave -Scope CurrentUser
Login-AzureRmAccount
Get-AzureRmSubscription

# Open new window with Core PowerShell
Get-AzureRmSubscription
```

It does nothing and throws this warning:
```
WARNING: Unable to acquire token for tenant '<TenantID>'
```

#### Full PowerShell

```powershell
# Full PowerShell
Get-Module -Name AzureRM -ListAvailable
```

```
    Directory: C:\Program Files\WindowsPowerShell\Modules


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     6.1.1      AzureRM
Script     5.4.1      AzureRM
Script     4.4.1      AzureRM
```

```powershell
Get-Module -Name AzureRM.Profile -ListAvailable
```

```
    Directory: C:\Program Files\WindowsPowerShell\Modules


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     5.1.0      AzureRM.profile                     {Disable-AzureRmDataCollection, Disable-AzureRmContextAuto...
Script     4.3.1      AzureRM.profile                     {Disable-AzureRmDataCollection, Disable-AzureRmContextAuto...
Script     3.4.1      AzureRM.profile                     {Disable-AzureRmDataCollection, Disable-AzureRmContextAuto...
```

#### Core PowerShell

```powershell
# Core PowerShell
Get-Module AzureRM.NetCore -ListAvailable
```

```
    Directory: C:\Users\iiric\Documents\PowerShell\Modules


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     0.11.0     AzureRM.Netcore
```

```powershell
Get-Module AzureRM.Profile.Netcore -ListAvailable
```

```
    Directory: C:\Users\iiric\Documents\PowerShell\Modules


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     0.11.0     AzureRM.Profile.Netcore             {Disable-AzureRmDataCollection, Disable-AzureRmContextAutosave, Enable-AzureRmDataCollection, Enable-AzureRmContextAutosave...}
```

### Environment Data

```powershell
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.17134.81
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17134.81
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

```powershell
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.1.0-preview.2
PSEdition                      Core
GitCommitId                    v6.1.0-preview.2
OS                             Microsoft Windows 10.0.17134
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by markcowl** [comment]
@iricigor It's unclear from your decription whether this has anything to do with context autosave, or this has to do with your permissions using the different login mechanisms.

The warning about tenants, for example, has nothign to do with context autosave.  Can you include the Debug log when executing these commands (set ```$DebugPreference="Continue"```) this captures the communication with  the server, including the subscriptioisn you have access to.

Once consequence of having context autosave turned on that might show up here is that yoiur default context may be from your rpevioius login.  Runnign Get-AzureRmContext (or the debug stream output) will indicate which user acocunt and context you are using.

**Comment by iricigor** [comment]
Hi Mark, 

Here is additional information how I was testing this. I will post debug log in next comment. The first letter in a row indicates if the step was done in (W) Windows PowerShell or in (C) PowerShell Core.

### Test 1
- (W) I open **Windows** PowerShell console and login with **myaccount1**. I run test command to confirm. 
- (W) Then I open another Windows PowerShell window just to confirm autologin. I run test command again, and it works fine. 
- (C) Then, I open PowerShell **Core** console and run the same test command. Now it fails because it cannot autologin.

Then I close all these windows and run a new test. 

### Test 2

- (C) I open PowerShell **Core** console and login with **myaccount2** which requires MFA. Running test command confirms login was ok. 
- (C) I open another Core console and run test command. It confirms autosave happened for new credentials.
- (W) Then I open **Windows** PowerShell (full CLR) console, run test command and it fails!

### Comments

- Test command in all cases was `(Get-AzureRmSubscription).Name`
- Windows PowerShell opens a mini-browser window to login, while PowerShell Core is asking me for device login with given code.

### My Conclusion

My conclusion was that autosave works fine, but ONLY within the same edition of PowerShell.  If I  open different edition of PowerShell, autologin is not working for me.

**Comment by vladimir-shcherbakov** [comment]
@markcowl  Please respond.

**Comment by markcowl** [comment]
@iricigor This is correct - the serialization of the TokenCache is different between the two cmdlet sets.  This will be resolved by #6000 ultimately.  For now, you can use ```Disable-AzureRmContextAutosave -Scope Process``` to keep this from happening for one or the other.
